app/src/main/java/com/example/android_level_3/FragmentSettings.kt

```diff
@@ -5,24 +5,110 @@ import androidx.fragment.app.Fragment
 import android.view.ViewGroup
-import androidx.navigation.fragment.findNavController
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.MutableLiveData
 import com.example.android_level_3.databinding.FragmentSettingsBinding
+import com.example.android_level_3.viewmodel.MainViewModel
+import com.google.android.material.snackbar.Snackbar
 
 class FragmentSettings : Fragment() {
 
     private lateinit var binding: FragmentSettingsBinding
 
+    private val viewModel: MainViewModel by activityViewModels()
+    private val sharedPreferences by lazy { requireActivity().getSharedPreferences(Const.PREFERENCES_SETTINGS,
+        AppCompatActivity.MODE_PRIVATE) }
+
+    private var connectionErrorSnackbar: Snackbar? = null
+    private var isVisibleProgressBar = MutableLiveData(false)
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         binding = FragmentSettingsBinding.inflate(inflater, container, false)
 
-        binding.btnSettingsViewContacts.setOnClickListener {
-            findNavController().navigate(R.id.action_fragmentSettings_to_fragmentContactsList)
-        }
+        setUserDataToUI()
+        setObservers()
+        setButtonListener()
 
+        if (sharedPreferences?.getBoolean(Const.PREFERENCES_AUTOLOGIN, false) == true) {
+            refreshUserDataAfterAutoLogin()
+        }
         return binding.root
+
+//      стало не нужным (в задании уроаня 4) ввиду использования TabLayout (пока оставить)
+//        binding.btnSettingsViewContacts.setOnClickListener {
+//            findNavController().navigate(R.id.action_fragmentSettings_to_fragmentContactsList)
+//        }
+    }
+
+    private fun setObservers() {
+        isVisibleProgressBar.observe(viewLifecycleOwner) { visibility ->
+            if (visibility) binding.progressBar.visibility = View.VISIBLE
+            else binding.progressBar.visibility = View.GONE
+        }
+
+        viewModel.authorisationResult.observe(viewLifecycleOwner) { serverResponse ->
+            if (serverResponse == null) {                                                           // показываем Snackbar с ошибкой и даем возможность перезапустить
+                connectionErrorSnackbar = createErrorConnectionSnackBar()
+                connectionErrorSnackbar?.show()
+            } else if (serverResponse.isSuccessful) {
+
+                // сохраняем обновленные данные AccessToken и RefreshToken (после авторизации)
+                sharedPreferences.edit().apply {
+                    putString(Const.PREFERENCES_ACCESS_TOKEN, serverResponse.body()?.data?.accessToken.toString())
+                    putString(Const.PREFERENCES_REFRESH_TOKEN, serverResponse.body()?.data?.refreshToken.toString())
+                }.apply()
+
+                viewModel.token.value = "Bearer ${serverResponse.body()?.data?.accessToken}"
```

Comment:
This action is not the responsibility of the UI, should be moved to the Data layer.